charts/kubeclarity/templates/scanner-template-configmap.yaml

```diff
@@ -109,10 +109,18 @@ data:
 {{- end}}
 {{- if index .Values "kubeclarity-runtime-scan" "httpsProxy" }}
             - name: NO_PROXY
+{{- if index .Values "kubeclarity-trivy-server" "enabled" }}
               value: {{ include "kubeclarity.name" . }}.{{ .Release.Namespace }}:{{ index .Values "kubeclarity-runtime-scan" "resultServicePort" }},{{ include "kubeclarity.grype-server.name" . }}.{{ .Release.Namespace }}:{{ index .Values "kubeclarity-grype-server" "servicePort" }},{{ (index .Values "kubeclarity-trivy-server" "service" "name") | default (include "trivy.fullname" (index .Subcharts "kubeclarity-trivy-server")) }}.{{ .Release.Namespace }}:{{ index .Values "kubeclarity-trivy-server" "service" "port" }}
+{{- else}}
+              value: {{ include "kubeclarity.name" . }}.{{ .Release.Namespace }}:{{ index .Values "kubeclarity-runtime-scan" "resultServicePort" }},{{ include "kubeclarity.grype-server.name" . }}.{{ .Release.Namespace }}:{{ index .Values "kubeclarity-grype-server" "servicePort" }}
```

Comment:
I think we should approach this a little differently to allow it to be extensible in the future when/if we add additional server types for other plugins. We can use some of the tools in the helm template to contruct a list in a variable at the top of this file like:
```
{{ $noproxy := list (print (include "kubeclarity.name" .) "." .Release.Namespace ":" (index .Values "kubeclarity-runtime-scan" "resultServicePort")) }}
{{ if index .Values "kubeclarity-grype-server" "enabled" }}
{{ $noproxy = append $noproxy (print (include "kubeclarity.grype-server.name" .) "." .Release.Namespace ":" (index .Values "kubeclarity-grype-server" "servicePort")) }}
{{ end }}
{{ if index .Values "kubeclarity-trivy-server" "enabled" }}
{{ $noproxy = append $noproxy (print ((index .Values "kubeclarity-trivy-server" "service" "name") | default (include "trivy.fullname" (index .Subcharts "kubeclarity-trivy-server"))) "." .Release.Namespace ":" (index .Values "kubeclarity-trivy-server" "service" "port")) }}
{{ end}}
```

and then here the template will become something like:

```
{{- if index .Values "kubeclarity-runtime-scan" "httpsProxy" }}
- name: NO_PROXY
   value: {{ join "," $noproxy }}
{{- end}}
```

it also means that we can avoid duplicating the logic like we do today.

Reply:
Hello @sambetts-cisco i added a function to generate the noProxy hosts to the helper template, hope that's how you expected me to do it.

Reply:
$noproxy is now evaluated without helper function.